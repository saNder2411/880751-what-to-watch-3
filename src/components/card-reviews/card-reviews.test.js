import React from 'react';
import renderer from 'react-test-renderer';
import CardReviews from './card-reviews.jsx';

const data = {
  id: 0,
  overviewData: {
    promoPoster: `bg-the-grand-budapest-hotel`,
    poster: `the-grand-budapest-hotel-poster`,
    previewPoster: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`
    ],
    rating: `10`,
    amountVoice: 100,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  detailsData: {
    director: `Steven Spielberg`,
    actors: [
      `Judi Dench`, `Robert De Niro`, `Leonardo DiCaprio`, `Morgan Freeman`, `Tom Hanks`,
    ],
    runtime: `1h 58m`,
    genre: `Drama`,
    release: 1989,
  },
  reviewsId: [879],
};

it(`Should CardReviews render correctly`, () => {
  const markup = renderer
    .create(<CardReviews data={data} />)
    .toJSON();

  expect(markup).toMatchSnapshot();
});
