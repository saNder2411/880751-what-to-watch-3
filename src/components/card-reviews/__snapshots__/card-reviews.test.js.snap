// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CardReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            January 1, 2567
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        10
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            January 1, 1967
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7
      </div>
    </div>
  </div>
</div>
`;
